### PR TITLE
Fixes #35100 - disable submit if no changes in roles edit

### PR DESCRIPTION
**Pull request by Ron-Lavi**

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
@stejskalleos JS & Ruby tests are green now

File: foreman_ansible.gemspec [hunk 1]
```diff
@@ -20,6 +20,6 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'acts_as_list', '~> 1.0.3'
   s.add_dependency 'deface', '< 2.0'
-  s.add_dependency 'foreman_remote_execution', '>= 4.4.0'
-  s.add_dependency 'foreman-tasks', '>= 5.2.0'
+  s.add_dependency 'foreman_remote_execution', '~> 7.0.0'
```
Comment (stejskalleos) [inline foreman_ansible.gemspec:23]: I really hope we won't break anything with this

File: .github/workflows/js_tests.yml [hunk 2]
```diff
@@ -12,13 +12,13 @@ jobs:
...
       matrix:
-        node-version: [10, 12]
+        node-version: [12]
     steps:
       - uses: actions/checkout@v2
       - uses: actions/checkout@v2
         with:
           repository: theforeman/foreman
-          ref: develop
+          ref: 3.3-stable
```
Comment (stejskalleos) [inline .github/workflows/js_tests.yml:21]: Awesome, it's nice to have green tests for cherry-picks too!

**Review by stejskalleos — CHANGES_REQUESTED** [review]
On Foreman 3.3 it doesn't work for me,
I'm getting error on Host > Ansible roles page:
```
Error!
Field 'ansibleVariables' doesn't exist on type 'AnsibleRole'
```

**Comment by Ron-Lavi** [comment]
@stejskalleos reordered the commits and changed the message of the last one,
ready for another round

**Review by stejskalleos — APPROVED** [review]
:green_apple:  **LGTM**

Thanks @Ron-Lavi, it's great improvement for `7.1-stable`

**Comment by stejskalleos** [comment]
Released in 7.1.1: https://github.com/theforeman/foreman_ansible/releases/tag/v7.1.1
RPM: https://github.com/theforeman/foreman-packaging/pull/8229
DEB: https://github.com/theforeman/foreman-packaging/pull/8230

**Pull request merged**